File: day20_FE_AT/EmpMan_phase2/tests/pages/SettingsPage.js
/**
 * Page Object Model – Settings Module
 * Tabs: Profile, Company, Notifications, Appearance, Security (Change Password), Roles & Permissions.
 */
class SettingsPage {
  constructor(page) {
    this.page = page;

    this.container     = page.locator('.settings');
    this.pageTitle     = page.locator('.dashboard-title').first();

    // Side-nav tabs
    this.profileTab      = page.locator('#settings-tab-profile');
    this.companyTab      = page.locator('#settings-tab-company');
    this.notificationsTab = page.locator('#settings-tab-notifications');
    this.appearanceTab   = page.locator('#settings-tab-appearance');
    this.securityTab     = page.locator('#settings-tab-security');
    this.rolesTab        = page.locator('#settings-tab-roles');

    // Profile form
    this.profileForm     = page.locator('#admin-profile-form');
    this.nameInput       = page.locator('[name="name"]').first();
    this.emailInput      = page.locator('[name="email"]').first();
    this.phoneInput      = page.locator('[name="phone"]').first();
    this.languageSelect  = page.locator('[name="language"]').first();
    this.timezoneSelect  = page.locator('[name="timezone"]').first();
    this.saveProfileBtn  = page.locator('#save-profile-btn');
    this.savedMessage    = page.locator('span', { hasText: /Saved successfully/i });

    // Security / Change password form
    this.currentPwdInput  = page.locator('[name="currentPassword"], [placeholder*="Current"], [placeholder*="current"]').first();
    this.newPwdInput      = page.locator('[name="newPassword"], [placeholder*="New password"], [placeholder*="new password"]').first();
    this.confirmPwdInput  = page.locator('[name="confirmPassword"], [placeholder*="Confirm"]').first();
    this.changePwdBtn     = page.locator('button', { hasText: /Change Password|Update Password/i }).first();

    // Theme toggle
    this.themeToggle      = page.locator('button', { hasText: /Toggle|Dark|Light|Theme/i }).first();

    // Sign out button inside settings
    this.signOutBtn       = page.locator('button', { hasText: /Sign Out/i });
  }

  async waitForLoad() {
    await this.container.waitFor({ state: 'visible', timeout: 15000 });
    await this.page.waitForTimeout(500);
  }

  async clickProfileTab() {
    await this.profileTab.click();
    await this.page.waitForTimeout(400);
  }

  async clickCompanyTab() {
    await this.companyTab.click();
    await this.page.waitForTimeout(400);
  }

  async clickNotificationsTab() {
    await this.notificationsTab.click();
    await this.page.waitForTimeout(400);
  }

  async clickAppearanceTab() {
    await this.appearanceTab.click();
    await this.page.waitForTimeout(400);
  }

  async clickSecurityTab() {
    await this.securityTab.click();
    await this.page.waitForTimeout(400);
  }

  async clickRolesTab() {
    await this.rolesTab.click();
    await this.page.waitForTimeout(400);
  }

  async updateProfileName(name) {
    await this.nameInput.fill('');
    await this.nameInput.fill(name);
  }

  async saveProfile() {
    await this.saveProfileBtn.click();
    await this.page.waitForTimeout(1000);
  }

  async isSavedMessageVisible() {
    return this.savedMessage.isVisible();
  }

  async clickThemeToggle() {
    if (await this.themeToggle.isVisible()) {
      await this.themeToggle.click();
      await this.page.waitForTimeout(400);
    }
  }

  async clickSignOut() {
    await this.signOutBtn.click();
    await this.page.waitForSelector('.login-page', { timeout: 10000 });
  }
}

module.exports = { SettingsPage };

File: day20_FE_AT/EmpMan_phase2/tests-java/src/main/java/com/ems/pages/SettingsPage.java
package com.ems.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

public class SettingsPage {
    private final Page page;

    public final Locator container;
    public final Locator profileTab;
    public final Locator companyTab;
    public final Locator notificationsTab;
    public final Locator appearanceTab;
    public final Locator securityTab;
    public final Locator rolesTab;
    
    // Profile form
    public final Locator nameInput;
    public final Locator emailInput;
    public final Locator phoneInput;
    public final Locator saveProfileBtn;
    public final Locator savedMessage;
    
    public final Locator signOutBtn;

    public SettingsPage(Page page) {
        this.page = page;
        this.container = page.locator(".settings-container, .settings-content");
        
        // Navigation tabs
        this.profileTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText("Profile Settings").setHasText(java.util.regex.Pattern.compile("(?i)profile")));
        this.companyTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText("Company"));
        this.notificationsTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText("Notifications"));
        this.appearanceTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText("Appearance"));
        this.securityTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText("Security"));
        this.rolesTab = page.locator("button, a").filter(new Locator.FilterOptions().setHasText(java.util.regex.Pattern.compile("(?i)roles")));

        // Profile Form Elements
        this.nameInput = page.locator("input[name='name'], input[placeholder*='Name']").first();
        this.emailInput = page.locator("input[name='email'], input[type='email']").first();
        this.phoneInput = page.locator("input[name='phone'], input[type='tel']").first();
        this.saveProfileBtn = page.locator("button").filter(new Locator.FilterOptions().setHasText("Save Changes"));
        this.savedMessage = page.locator(".alert-success, .toast-success");
        
        this.signOutBtn = page.locator("button").filter(new Locator.FilterOptions().setHasText(java.util.regex.Pattern.compile("(?i)sign out")));
    }

    public void waitForLoad() {
        container.waitFor();
    }

    public void clickRolesTab() {
        if(rolesTab.isVisible()) rolesTab.click();
    }
    
    public void clickSecurityTab() {
        if(securityTab.isVisible()) securityTab.click();
    }
    
    public void clickProfileTab() {
        if(profileTab.isVisible()) profileTab.click();
    }
}
